### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by angtroake**
Unit tests for StarlingX Configuration Management API. This pull adds unit tests for all remaining API endpoints.

Edit:
Relies-on: #4

### CODE REVIEW

**Review by steven-webster — COMMENTED** [review]
I get an error running this test.  Are you seeing the same?

$ go test github.com/gophercloud/gophercloud/starlingx/inventory/v1/storagebackends/testing/
--- FAIL: TestUpdateStorageBackend (0.00s)
    convenience.go:300: Expected JSON:
        [
          {
            "op": "replace",
            "path": "/capabilities",
            "value": {}
          },
          {
            "op": "replace",
            "path": "/confirmed",
            "value": false
          }
        ]
    convenience.go:308: Actual JSON:
        [
          {
            "op": "replace",
            "path": "/confirmed",
            "value": false
          },
          {
            "op": "replace",
            "path": "/capabilities",
            "value": {}
          }
        ]
    convenience.go:39: Failure in http_responses.go, line 94: The generated JSON structure differed.
FAIL
FAIL	github.com/gophercloud/gophercloud/starlingx/inventory/v1/storagebackends/testing	0.006s

File: starlingx/inventory/v1/storagebackends/testing/fixtures.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		w.WriteHeader(http.StatusNoContent)
+	})
+}
+
+func HandleStorageBackendCreationSuccessfully(t *testing.T, response string) {
+	th.Mux.HandleFunc("/storage_backend", func(w http.ResponseWriter, r *http.Request) {
+		th.TestMethod(t, r, "POST")
+		th.TestHeader(t, r, "X-Auth-Token", client.TokenID)
+		th.TestJSONRequest(t, r, `{
+          "backend": "ceph",
```
Comment (steven-webster) [inline starlingx/inventory/v1/storagebackends/testing/fixtures.go]: indentation looks a bit off

File: starlingx/inventory/v1/storagebackends/testing/fixtures.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+func HandleStorageBackendUpdateSuccessfully(t *testing.T) {
+	th.Mux.HandleFunc("/storage_backend/cebe7a5e-7b57-497b-a335-6e7cf93e98ee", func(w http.ResponseWriter, r *http.Request) {
+		th.TestMethod(t, r, "PATCH")
+		th.TestHeader(t, r, "X-Auth-Token", client.TokenID)
+		th.TestHeader(t, r, "Accept", "application/json")
+		th.TestHeader(t, r, "Content-Type", "application/json")
+		th.TestJSONRequest(t, r, `[{"path": "/capabilities", "value": {}, "op": "replace"},
+            {"op": "replace", "path": "/confirmed", "value": false}]`)
```
Comment (steven-webster) [inline starlingx/inventory/v1/storagebackends/testing/fixtures.go]: indentation looks a bit off

**Review by steven-webster — COMMENTED** [review]
I ran through all of these tests in my environment.  All look good except the storagebackend failure

**Comment by angtroake** [comment]
> I ran through all of these tests in my environment. All look good except the storagebackend failure

I don't have that issue on my end. storagebackend returns OK.

**Comment by donpenney** [comment]
I also had this failure, where the JSON content appears to be the same, but ordered differently:

?       github.com/gophercloud/gophercloud/starlingx/inventory/v1/storagebackends       [no test files]
--- FAIL: TestUpdateStorageBackend (0.00s)
    convenience.go:300: Expected JSON:
        [
          {
            "op": "replace",
            "path": "/capabilities",
            "value": {}
          },
          {
            "op": "replace",
            "path": "/confirmed",
            "value": false
          }
        ]
    convenience.go:308: Actual JSON:
        [
          {
            "op": "replace",
            "path": "/confirmed",
            "value": false
          },
          {
            "op": "replace",
            "path": "/capabilities",
            "value": {}
          }
        ]
    convenience.go:39: Failure in http_responses.go, line 94: The generated JSON structure differed.
FAIL
FAIL    github.com/gophercloud/gophercloud/starlingx/inventory/v1/storagebackends/testing       0.016s

**Comment by donpenney** [comment]
Everything now passes with the new updates. I guess I tried it out a few minutes too early :)

File: starlingx/inventory/v1/cpus/results.go [hunk 3]
```diff
@@ -42,15 +42,15 @@ type CPU struct {
 	// Processor is the processor package number.  A host usually has 1 or 2
 	// processors but that number could be higher.  Each is usually associated
 	// to a seperate NUMA socket.
-	Processor int `json:"processor"`
+	Processor int `json:"numa_node"`
```
Comment (donpenney) [inline starlingx/inventory/v1/cpus/results.go:45]: Why are these being changed? Was the mapping wrong?

File: starlingx/inventory/v1/interfaceDataNetworks/results.go [hunk 4]
```diff
@@ -50,11 +50,11 @@ type InterfaceDataNetwork struct {
 
 	// DataNetworkUUID defines the system assigned unique UUID of the associated
 	// data network.
-	DataNetworkUUID string `json:"network_uuid"`
+	DataNetworkUUID string `json:"datanetwork_uuid"`
```
Comment (donpenney) [inline starlingx/inventory/v1/interfaceDataNetworks/results.go:53]: Same question... was the mapping wrong?

**Comment by steven-webster** [comment]
storagebackends test passes for me now too

File: starlingx/inventory/v1/cpus/results.go (hunk 3 above)
Comment (angtroake) [reply starlingx/inventory/v1/cpus/results.go:45]: system --debug has field as numa_node

File: starlingx/inventory/v1/interfaceDataNetworks/results.go (hunk 4 above)
Comment (angtroake) [reply starlingx/inventory/v1/interfaceDataNetworks/results.go:53]: The API reference and system --debug has it as datanetwork_uuid (https://docs.starlingx.io/api-ref/config/api-ref-sysinv-v1-config.html?expanded=shows-information-about-a-specific-interface-data-network-detail#interface-data-networks)

File: starlingx/inventory/v1/osds/requests.go [hunk 5]
```diff
@@ -89,7 +89,7 @@ func (opts *OSDOpts) fixUnits() {
 
 // Create accepts a CreateOpts struct and creates a new volume group using the
 // values provided.
-func Create(c *gophercloud.ServiceClient, opts OSDOpts) (r CreateResult) {
+func Create(c *gophercloud.ServiceClient, hostid string, opts OSDOpts) (r CreateResult) {
```
Comment (donpenney) [inline starlingx/inventory/v1/osds/requests.go]: What is the reason for this change? Changes don't appear to have been made to callers of this function elsewhere

Comment (angtroake) [reply starlingx/inventory/v1/osds/requests.go]: This was reverted
